Prepare CHANGELOG.md for 1.50.0 (#2300)
open-telemetry/opentelemetry-java-contrib: CHANGELOG.md

## Unreleased

Note: This release broadly applies some style guidelines across the repository. As a result,
some classes that were visible might be package/private. Other non-final classes may now
be final. See
[#2182](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2182)
and
[#2210](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2210)
and
[#2212](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2212)
and
[#2213](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2212)

> for examples and details. These changes are not expected to be breaking, so please open

Reviewer: ```suggestion
for examples and details. These changes are not expected to break anyone, so please open
```

an issue if this causes problems.

### Baggage processor

> - Add baggage before it can be exported

Reviewer: ```suggestion
- Move baggage processor to the front of the processor list
```

  ([#2152](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2152))

> - Add declarative config for baggage

Reviewer: ```suggestion
- Add declarative configuration support
```

  ([#2031](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2031))

### Disk buffering

- Catching IllegalStateException in case of failed deserialization
  ([#2157](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2157))

> - Apply final to public API classes where possible - disk-buffering

Reviewer: ```suggestion
- Apply final to public API classes where possible
```

  ([#2216](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2216))

> - Handle empty attribute values in disk buffering

Reviewer: ```suggestion
- Handle empty attribute values
```

  ([#2268](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2268))

### Inferred spans

> - Add dynamically changing the inferred span interval

Reviewer: ```suggestion
- Support dynamically changing the inferred span interval
```

  ([#2153](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2153))

### JMX scraper

> - Implement stable service.instance.id

Reviewer: ```suggestion
- Implement stable `service.instance.id`
```

  ([#2270](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2270))

### Kafka exporter

- Add Kafka connectivity error handling
  ([#2202](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2202))

### OpAMP client

> - Move important user-facing classes out of 'internal' package.

Reviewer: ```suggestion
- Move important user-facing classes out of 'internal' package
```

Author: We had a conflicting standard here internally...so this is a hard one for me to break. I'll try! 😁

  ([#2249](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2249))
- Exponential backoff retries on http connection failures
  ([#2274](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/2274))

### Span stack traces

> - Declarative config: support Span stacktrace

Reviewer: ```suggestion
- Add declarative configuration support
```
